gammapy/estimators/energydependence.py

```diff
@@ -0,0 +1,250 @@
+def weighted_chi2_parameter_results(table_edep, parameter="sigma"):
+    """Calculate the weighted chi2 value for the parameter of interest"""
+
+    values = table_edep[f"{parameter}"][1:]
+    errors = table_edep[f"{parameter}_err"][1:]
+
+    weights = 1 / errors**2
+    avg = np.average(values, weights=weights)
+
+    chi2_value = np.sum((values - avg) ** 2 / errors**2)
+    df = len(values) - 1
+    sigma_value = ts_to_sigma(chi2_value, df)
+
+    chi2_result = {}
+    chi2_result[f"chi2 {parameter}"] = [chi2_value]
+    chi2_result["df"] = [df]
+    chi2_result["significance"] = [sigma_value]
+
+    return chi2_result
+
+
+class EnergyDependenceEstimator(Estimator):
+    """Test if there is any energy-dependent morphology in a map dataset for a given set of energy bins.
+
+    Parameters
+    ----------
+    energy_edges : `~astropy.units.Quantity`
+        Energy edges for the energy-dependence test.
+    model : `~gammapy.modeling.model.SkyModel`
+        Source model kernel.
+    selection_optional : list of str
+        Which additional quantities to estimate. Available options are:
+            * "src-sig": the significance above the background is
+            calculated in each energy band
+
+    Examples
+    --------
+
+    """
+
+    tag = "EnergyDependenceEstimator"
+    _available_selection_optional = ["src-sig"]
+
+    def __init__(self, energy_edges, model, fit=None, selection_optional=None):
+
+        self.energy_edges = energy_edges
+        self.model = model
+        self.num_energy_bands = len(self.energy_edges) - 1
+
+        if fit is None:
+            fit = Fit(optimize_opts={"print_level": 1})
+
+        self.fit = fit
+        self.selection_optional = selection_optional
+
+    def estimate_source_significance(self, dataset):
+        """Estimate the significance of the source above the background.
+
+        Parameters
+        ----------
+        dataset : `~gammapy.datasets.MapDataset`
+            Input dataset to use.
+
+        Returns
+        -------
+        table_bkg_src : `~astropy.table.Table`
+            Table with the results of the null hypothesis with no source, and alternative
+            hypothesis with the source added in. Entries are:
+            * "Emin" : the minimum energy of the energy band
+            * "Emax" : the maximum energy of the energy band
+            * "delta ts" : difference in ts
+            * "df" : the degrees of freedom between null and alternative hypothesis
+            * "significance" : significance of the result
+        """
+
+        dataset.mask_fit = dataset.counts.geom.energy_mask(
+            energy_min=self.energy_edges[0], energy_max=None
+        )
+
+        # Calculate the dataset for each energy slice
+        slices_src = Datasets()
+        for emin, emax in zip(self.energy_edges[:-1], self.energy_edges[1:]):
+            sliced_src = dataset.slice_by_energy(emin, emax)
+            bkg_sliced_model = FoVBackgroundModel(dataset_name=sliced_src.name)
```

Comment:
it might be useful to fit the background as well. Especially at high energies, the source significance varies significantly depending on the background model